openmls/src/tree/secret_tree.rs

```diff
@@ -352,16 +350,18 @@ impl SecretTree {
 
     /// Returns a mutable reference to a specific SenderRatchet. The
     /// SenderRatchet needs to be initialized.
-    fn ratchet_mut(&mut self, index: LeafNodeIndex, secret_type: SecretType) -> &mut SenderRatchet {
+    fn ratchet_mut(
+        &mut self,
+        index: LeafNodeIndex,
+        secret_type: SecretType,
+    ) -> Result<&mut Option<SenderRatchet>, SecretTreeError> {
```

Comment:
Just a nit, but if you return `Option<&mut SenderRatchet>` here instead, you save the `.as_mut()` in the places where you call this. 

I know you don't want to refactor the secret tree generally, but there are a few places you touch where we get the secrets "manually" via `get_mut` instead of using this function. Might be worth just using this function everywhere instead.